src/Decoder.c

```diff
@@ -4804,8 +4804,12 @@ static ZyanStatus ZydisDecodeInstruction(ZydisDecoderContext* context,
                         ZYAN_MEMCPY(&instruction->accessed_flags, &flags->action,
                             sizeof(flags->action));
 
-                        instruction->flags_read = flags->flags_read;
-                        instruction->flags_written = flags->flags_written;
+                        instruction->cpu_flags_read = flags->flags_read & 0x003FFFFF;
+                        instruction->cpu_flags_written = flags->flags_written & 0x003FFFFF;
+                        instruction->fpu_flags_read = 
+                            (ZydisFPUFlags)((flags->flags_read & 0x03C00000) >> 22);
+                        instruction->fpu_flags_written =
+                            (ZydisFPUFlags)((flags->flags_written & 0x03C00000) >> 22);
```

Comment:
I'm a bit confused: what constant is going to correspond to that after the shift? `ZYDIS_CPUFLAG_C0` is defined as `22`, so it doesn't appear to be expecting a shift.

Reply:
You are right, for now there are no corresponding defines (the CPUFLAG ones are for backwards compatibility - e.g. to use as an index in the accessed_flags array). I will create additional FPUFLAG ones for improved usability. The CPUFLAG ones (only C0..C3) will be removed in the breaking release.